q2_feature_table/tests/test_heatmap.py

```diff
@@ -171,6 +194,57 @@ def test_munge_metadata_sort_samples(self):
                            index=exp_idx)
         assert_frame_equal(exp, obs)
 
+    def test_munge_feature_metadata_simple(self):
+        feature_md = qiime2.CategoricalMetadataColumn(
+            pd.Series(['peanut', 'dog'], name='species',
+                      index=pd.Index(['O1', 'O2'], name='id')))
+        obs = _munge_feature_metadata(feature_md, self.table, 'both')
+
+        exp = pd.DataFrame(
+            [[0, 10], [10, 12], [10, 11]], columns=['peanut', 'dog'],
+            index=pd.Index(['S1', 'S2', 'S3']))
+        assert_frame_equal(exp, obs)
+
+    def test_munge_feature_metadata_ids_different_order(self):
+        feature_md = qiime2.CategoricalMetadataColumn(
+            pd.Series(['dog', 'peanut'], name='species',
+                      index=pd.Index(['O2', 'O1'], name='id')))
+        obs = _munge_feature_metadata(feature_md, self.table, 'both')
+
+        exp = pd.DataFrame(
+            [[0, 10], [10, 12], [10, 11]], columns=['peanut', 'dog'],
+            index=pd.Index(['S1', 'S2', 'S3']))
+        assert_frame_equal(exp, obs)
+
+    def test_munge_feature_metadata_missing_features(self):
+        feature_md = qiime2.CategoricalMetadataColumn(
+            pd.Series(
+                ['dog'], name='species', index=pd.Index(['O2'], name='id')))
+        with self.assertRaisesRegex(ValueError, 'not present.*O1'):
+            _munge_feature_metadata(feature_md, self.table, 'both')
+
+    def test_munge_feature_metadata_is_superset(self):
+        feature_md = qiime2.CategoricalMetadataColumn(
+            pd.Series(['peanut', 'dog', 'cujo'], name='species',
+                      index=pd.Index(['O1', 'O2', 'O3'], name='id')))
+        obs = _munge_feature_metadata(feature_md, self.table, 'both')
+
+        exp = pd.DataFrame(
+            [[0, 10], [10, 12], [10, 11]], columns=['peanut', 'dog'],
+            index=pd.Index(['S1', 'S2', 'S3']))
+        assert_frame_equal(exp, obs)
+
+    def test_munge_feature_metadata_sort_samples(self):
```

Comment:
Please add a test to cover the case when there are duplicate feature values (as described above).

Reply:
👍